src/css3/new_tokenizer.rs

```diff
@@ -89,6 +101,148 @@ impl<'stream> Tokenizer<'stream> {
         Number::new(kind, value.parse().expect("failed to parse number"))
     }
 
+    /// 4.3.4. [Consume an ident-like token](https://www.w3.org/TR/css-syntax-3/#consume-ident-like-token)
+    ///
+    /// Returns: `<ident-token>`, `<function-token>`, `<url-token>`, or `<bad-url-token>`.
+    pub fn consume_ident_like_seq(&mut self) -> Token {
+        let value = self.consume_ident();
+
+        if value == "url" && self.stream.current_char().utf8() == '(' {
+            // consume '('
+            self.stream.read_char();
+            self.consume_whitespace();
+
+            if self.is_any_of(vec!['"', '\'']) {
+                return Token::Function(value);
+            }
+
+            return self.consume_url();
+        } else if self.stream.current_char().utf8() == '(' {
+            // consume '('
+            self.stream.read_char();
+            return Token::Function(value);
+        }
+
+        Token::Ident(value)
+    }
+
+    /// 4.3.6. [Consume a url token](https://www.w3.org/TR/css-syntax-3/#consume-a-url-token)
+    ///
+    /// Returns either a `<url-token>` or a `<bad-url-token>`
+    pub fn consume_url(&mut self) -> Token {
+        let mut url = String::new();
+
+        self.consume_whitespace();
+
+        loop {
+            if self.stream.current_char().utf8() == ')' {
+                println!("here..");
```

Comment:
Just adding a reminder to remove this `println` later, assuming it was being used for debugging